### PR TITLE
Ports SMES Human Mutation

### PR DESCRIPTION
**Pull request by Iamgoofball**
Ports the SMES Human mutation from Goonstation.

Demonstration:

### CODE REVIEW

**Comment by KorPhaeron** [comment]
I'd prefer it have some downside or something rather than genetics churning out passive immunity at no cost.

**Comment by MMMiracles** [comment]
port goon genetics when

**Comment by ToadStar** [comment]
It also has a Biostability variable that goes down when you get more powers, the better powers costing more biostability. Changelings are immune to this and there's a trait you can get at character setup that gives you a bonus 20 max biostability.

**Comment by KorPhaeron** [comment]
@ToadStar 

Yeah that sounds good. With semi-recent changes to our genetics you don't even have to reroll your powers when you inject, you just keep piling on more buffs

**Comment by MMMiracles** [comment]
If I remember correctly, passing max stability essentially turns you into an unrecognizable pile of goop and you get kicked out of the body.

**Comment by ToadStar** [comment]
Goon genetics works very neatly; You roll 5 different powers and disabilities at roundstart, and you can get those activated after genetics researches 'em, needing your DNA sample or someone else with that power, and you can get powers that aren't in your genome, but at a higher biostability cost iirc. And looking at goon genetics codewise it should be easyish to port.

**Comment by xxalpha** [comment]
Human SMES should need to constantly suck up electricity or die.

**Comment by KazeEspada** [comment]
Make attack_hand on anything with conduct cause a short circuit and do burn damage.

**Comment by Fox-McCloud** [comment]
>port goon genetics when

This; mostly the research+discovery system; it really gates power abuse and eliminates 'push button, receive power randomly'.

And yeah, they have a thing where you take damage if you have too many genes active at once, and if you have way too many, you can out and out gib (keep in mind this is veryyyy hard to use offensively against someone though because there aren't full injectors on Goon).

**Comment by Core0verload** [comment]
"SMES" has the second S in it for a reason. It stands for "Storage". 

So make SMES humans store charge, and display the SMES icon only when charged/overcharged. Let them use their charge for nutrition, maybe charging cells, etc. Being at 0 charge shouldn't be harmful, but overcharge must deal some burn damage per tick, and at some critical charge point it should simply explode the SMES man from the inside. So it wouldn't be "lol tesla cant hurt me", but will allow you to survive a few shocks as long as you don't abuse it.

**Comment by WJohn** [comment]
Those are some awful sprites :-1: but cool idea otherwise. And no I can't give you sprites right now I have no internet at home.

**Comment by Iamgoofball** [comment]
They look better when overlayed OVER the person

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -89,6 +89,8 @@
 
 
 /mob/living/carbon/electrocute_act(shock_damage, obj/source, siemens_coeff = 1, override = 0, tesla_shock = 0)
+	if(dna && dna.check_mutation(SMES_HUMAN))
```
Comment (Razharas) [inline code/modules/mob/living/carbon/carbon.dm]: All species have siemens coefficient that is like that on gloves
Just change it to 0 on mut_gain and to initial on mut_lose

Comment (Core0verload) [inline code/modules/mob/living/carbon/carbon.dm]: And then shit will happen on species change.

Comment (PKPenguin321) [inline code/modules/mob/living/carbon/carbon.dm]: what species has a different siemens coefficient?

Comment (RemieRichards) [inline code/modules/mob/living/carbon/carbon.dm]: it would just change it to the initial of the new species, which technically isn't wrong @Core0verload

Comment (Core0verload) [inline code/modules/mob/living/carbon/carbon.dm]: Yes, but then species change would remove any shock protection from the mutation. A good wy to get +1 issue nobody would ever fix.

Comment (Razharas) [inline code/modules/mob/living/carbon/carbon.dm]: Technically we dont have a clear way of asserting should species things override mutations or should mutations override species things
THus any behaviour is for now acceptable

**Comment by KorPhaeron** [comment]
Find some way to give it a downside or something instead of free passive buffs

**Comment by Razharas** [comment]
Mutations are passive buffs your argument is invalid

**Comment by Fox-McCloud** [comment]
>Mutations are passive buffs

Yeah, pretty much. 

Would be better to just have something in place that prevents excessive power stacking, rather than putting downsides to specific powers.

**Comment by KorPhaeron** [comment]
Port goon genetics first

**Pull request closed without merging**